swap_meet/vendor.py

```diff
@@ -1,2 +1,63 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+        
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        self.inventory.remove(item)
+        return item
+
+    def get_by_category(self, category=""):
+        items_by_category = []
+        for item in self.inventory:
+            if item.category == category:
+                items_by_category.append(item)
+        return items_by_category
+
+    def swap_items(self, friend, my_item, their_item):
+        if my_item not in self.inventory or their_item not in friend.inventory:
+            return False
+        self.inventory.remove(my_item)
+        friend.inventory.append(my_item)
+        friend.inventory.remove(their_item)
+        self.inventory.append(their_item)
+        return True
+
+
+    def swap_first_item(self, friend):
+        if len(self.inventory) == 0 or len(friend.inventory) == 0:
+            return False
+        return self.swap_items(friend,self.inventory[0],friend.inventory[0])
+        
+    def get_best_by_category(self, category):
+        items_by_category = self.get_by_category(category)
```

Comment:
This definitely works and I do like that it uses a helper method you wrote! There is a more straightforward way to accomplish this that only uses one for-loop, something like:
```python
        highest = 0
        best_item = None

        for item in self.inventory:
            if item.category == category and item.condition > highest:
                best_item = item
                highest = item.condition

        return best_item
```